packages/daemon/src/app/controller/pipeline.ts

```diff
@@ -45,19 +45,16 @@ export class PipelineController extends BaseEventController {
       if (parsedConfig[field]) {
         const pkg = await this.pluginManager.fetch(parsedConfig[field]);
         return this.pluginManager.findOrCreateByPkg(pkg);
-      } else {
-        return undefined;
       }
     };
-    const plugins = [];
-    await Promise.all(constants.PLUGINS.map(async (pluginType) => {
+    const plugins = (await Promise.all(constants.PLUGINS.map(async (pluginType) => {
       const plugin = await createPlugin(pluginType);
       if (plugin) {
-        plugins.push(plugin);
         parsedConfig[`${pluginType}Id`] = plugin?.id;
         parsedConfig[pluginType] = plugin?.name;
+        return plugin;
       }
-    }));
+    }))).filter((plugin) => plugin);
```

Comment:
It's good to exactly define a function `const isNotUndefined = (f) => f !== undefined` for this?

Reply:
```js
const plugins = (await Promise.all(constants.PLUGINS.map(createPlugin)))
  .filter(isNotUndefined)
  .map((plugin) => {
    parsedConfig[`${pluginType}Id`] = plugin?.id;
    parsedConfig[pluginType] = plugin?.name;
    return plugin
  });

Reply:
Otherwise LGTM :)